src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -252,6 +477,215 @@ private TemporaryPoint createNewPoint(final Coordinate coordinate,
+            final List<Line> closedInnerLines)
+    {
+        final MultiMap<Integer, Integer> outerToInnerIntersectionMap = new MultiMap<>();
+
+        if (!closedOuterLines.isEmpty() && !closedInnerLines.isEmpty())
+        {
+            for (int innerIndex = 0; innerIndex < closedInnerLines.size(); innerIndex++)
+            {
+                final Line inner = closedInnerLines.get(innerIndex);
+                for (int outerIndex = 0; outerIndex < closedOuterLines.size(); outerIndex++)
+                {
+                    final Line outer = closedOuterLines.get(outerIndex);
+
+                    // Make sure we're looking at the same country, not just intersection
+                    if (fromSameCountry(outer, inner) && outer.intersects(new Polygon(inner)))
+                    {
+                        outerToInnerIntersectionMap.add(outerIndex, innerIndex);
+                    }
+                }
+            }
+        }
+
+        return outerToInnerIntersectionMap;
+    }
+
+    /**
+     * @param one
+     *            The first {@link Line}
+     * @param two
+     *            The second {@link Line}
+     * @return {@code true} if both lines belong to the same country
+     */
+    private boolean fromSameCountry(final Line one, final Line two)
+    {
+        final Optional<IsoCountry> countryOne = ISOCountryTag.first(one);
+        final Optional<IsoCountry> countryTwo = ISOCountryTag.first(two);
+        if (countryOne.isPresent() && countryTwo.isPresent())
+        {
+            return countryOne.get().equals(countryTwo.get());
+        }
+        return false;
+    }
+
+    /**
+     * @param relationIdentifier
+     *            The {@link Relation} identifier whose members we're processing
+     * @param members
+     *            The list of {@link RelationMember} we want to group
+     * @param closed
+     *            {@code true} if we want to find closed member lines, false otherwise
+     * @return a subset of the given {@link RelationMember}s, either closed or non-closed
+     */
+    private List<RelationMember> generateMemberList(final long relationIdentifier,
+            final List<RelationMember> members, final boolean closed)
+    {
+        if (members != null && !members.isEmpty())
+        {
+            // Check if all outer ways are closed. If there are any that aren't closed, we need to
+            // build LineRings and fix the gaps
+            return members.stream().filter(member ->
+            {
+                final Line line = this.rawAtlas.line(member.getEntity().getIdentifier());
+                if (line == null)
+                {
+                    logger.error("Line member {} for Relation {} is not the in raw Atlas.",
+                            member.getEntity().getIdentifier(), relationIdentifier);
+                }
+
+                if (!closed)
+                {
+                    return line == null || !line.isClosed();
```

Comment:
So `null` `Line` objects should be permitted?

Reply:
Good catch! The original intent was to allow to process members that may have been added on the fly (part of the relation, but not in the raw Atlas - only in the temporary relation member change set version). However, looking again, I've refactored this in a such a way this can never happen. Will not allow null members here.